transformer_engine/pytorch/transformer.py

```diff
@@ -556,13 +772,23 @@ def forward(
         # core attention computation
         # ==================================
 
-        if checkpoint_core_attention:
-            context_layer = self._checkpointed_core_attention_forward(
-                query_layer, key_layer, value_layer, attention_mask
+        if self.use_flash_attention:
+            query_layer, key_layer, value_layer = [x.transpose(0,1).contiguous()
```

Comment:
This actually should be part of the FlashAttention forward function, that multiheadattention module should not really know whether the DotProductAttention uses FlashAttention or something else.

Reply:
We need to have a contract with the user of `DotProductAttention` class what its inputs are.